Add GSoC site (#110)
neuroinformatics-unit/neuroinformatics-unit.github.io: docs/source/get-involved/gsoc/projects_2025.md

Reviewer: I think this very large list of cards is hard to parse, and will become much harder to parse once we add more projects. Maybe there should be a card per repository, that then links to a list of projects? Or maybe even a much smaller card per project that links to the full description?

Author: Yeah agreed.

I made a card per repo as you suggested, which link to lists of projects where the project descriptions are collapsible - hopefully that makes it easier to parse?

# GSoC NIU Projects 2025

Below you can find our list of projects for GSoC 2025 (aka our "Ideas List"). 

> <!-- There are three main milestones in a GSoC project: the community bonding period, the start of coding, and the coding end.

Author: Question for reviewer(s):

Is this commented out section worth keeping?

At first I thought it was useful information, but then I think it may be repetitive with the GSoC docs. It feels like a maintenance burden to make sure this is updated and in sync with them if it changes for other years.

# GSoC NIU Projects 2025

Below you can find our list of projects for GSoC 2025 (aka our "Ideas List"). 

A project can be one of three sizes: small (90 h), medium (175 h) or large  large (350 h). The standard coding period is 12 weeks for medium and large projects, and 8 weeks for small projects. 

However, GSoC contributors can request in their proposal up to a 22-week coding period, if they know they may have other commitments or certain weeks when they will not be able to work full time on their GSoC project. During the project preparation period (called "community bonding period"), both the GSoC contributor and the mentors will agree on a schedule and sign off on it.

If you are interested in any of these projects, feel free to reach out by opening a new topic on [Zulip](https://neuroinformatics.zulipchat.com/), and tagging the potential mentors.

::::{grid} 1 1 1 1
:gutter: 4

:::{grid-item-card} {fas}`video;sd-text-primary` Support for Kalman filters in `movement`
<!-- Description -->

> Markerless pose estimation tools, such as [DeepLabCut](https://www.mackenziemathislab.org/deeplabcut) and [SLEAP](https://sleap.ai/), have revolutionised the study of animal behaviour. However, there is currently no user-friendly, general-purpose approach for processing and analysing the trajectories generated by these popular tools. To fill this gap, we're developing [`movement`](https://movement.neuroinformatics.dev/), an open-source Python package that provides a unified data analysis interface across pose estimation frameworks. 

Reviewer: I would emphasise the deep learning aspect here.

**Required skills**

Experience with Python and [PyTorch](https://pytorch.org/).

**Nice-to-haves**
- Experience developing or using [napari](https://napari.org/) plugins.
- Experience with computer vision applications, particularly pose estimation and any-point tracking approaches.

**Potential mentors**

- [@sfmig](https://github.com/sfmig)
- [@niksirbi](https://github.com/niksirbi)

**Further reading**
<!-- The best pages include links to more detailed descriptions and related materials for each project. They might even include actual use cases! -->
- [cotracker3 paper and code](https://cotracker3.github.io/)
- [TAPIR paper and code](https://deepmind-tapir.github.io/blogpost.html)
- [napari usage tutorials](https://napari.org/stable/tutorials/index.html) and [contributing guide](https://napari.org/stable/developers/contributing/index.html).
- [napari Plugin documentation](https://napari.org/stable/plugins/index.html), particularly the section on [Building a plugin](https://napari.org/stable/plugins/building_a_plugin/index.html).
:::

:::{grid-item-card} {fas}`video;sd-text-primary` Improve `cellfinder`'s classification algorithm
<!-- Description -->

> BrainGlobe is a community-driven suite of open-source Python tools. The BrainGlobe tools are widely used to process, analyse and visualise images of brains (and other related data) in neuroscientific research.

Reviewer 2: Is it worth emphasising the use of a common coordinate space here?

The BrainGlobe `cellfinder` tool is used to detect cells in large whole-brain images. It uses signal processing to find possible cell candidates and passes them to a customisable classifier to split the candidates into cells and no-cells. `cellfinder` relies heavily on `pytorch` and `keras`.

> `cellfinder` currently uses a residual neural network (ResNet) to classify cell candidates, and Deep Learning networks have progressed since. In this project, you will explore newer classification algorithms as alternatives to ResNet, and see whether they work better and/or faster than the existing implementation.

Reviewer 2: ```suggestion
`cellfinder` currently uses a residual neural network (ResNet) to classify cell candidates, and Deep Learning network architectures have progressed since. In this project, you will explore newer architectures for the classification network as alternatives to ResNet, and see whether they work better and/or faster than the existing implementation.
```

**Deliverables**
<!-- Goals, or expected status after Community Bonding Period, Start of Coding, End of Coding. Stretch goals? -->
The expected deliverables include:

> - A Python implementation of at least one new Deep Learning network in `cellfinder`

Reviewer 2: ```suggestion
- A Python implementation of at least one new Deep Learning network architecture in `cellfinder`
```

> - Tests to cover any added functionality.

Reviewer 2: ```suggestion
- Quantitative comparison between the current and new architecture
- Tests to cover any added functionality.
```
